app/main.py

```diff
@@ -1 +1,30 @@
-# write your code here
+import json
+import decimal
+
+
+def calculate_profit(file_name):
+    with open(file_name) as json_file:
+        trades = json.load(json_file)
+
+        earned_money = 0
+        matecoin_account = 0
+
+        for trade in trades:
+            price = decimal.Decimal(trade["matecoin_price"])
+            if trade["bought"] is not None:
+                earned_money -= decimal.Decimal(trade["bought"]) * price
+                matecoin_account += decimal.Decimal(trade["bought"])
+            if trade["bought"] is None:
```

Comment:
It is better to be sure that `if trade["sold"]` rather then `if trade["bought"] is None`